pkgs/applications/version-management/gogs/default.nix.patch

```diff
@@ -0,0 +1,22 @@
+2,3c2
```

Comment:
@kamilchm Oh, nice! But yes `allowGoReference` is an argument of `buildGoPackage` and `lib.overrideDerivation` allows only to custom the derivation attributes.

Is there some kind of override mechanism in Nix that would allow to customize the arguments to the `buildGoPackage` call?

An idea would be to tweak the `generated.nix` like this:

```nix
{ stdenv, lib, goPackages, fetchgit, fetchhg, fetchbzr, fetchsvn, args }:

with goPackages;

buildGoPackage (rec {
  name = "go2nix-${version}";
  # ...
} // args)
```

And call it like this:

```nix
callPackage ./generated.nix {
  args = {
    allowGoReference = true;
    # ...
  };
}
```

Or even allow to pass these arguments in the top-level function instead of the `args` set, like @cstrahan suggested, if I understood well? Something like the following example would avoid listing explicitly all possible overridable attributes, but I don't like this `filterAttrs` call, maybe there's a better way to get all the **remaining** args).

```nix
{ stdenv, lib, goPackages, fetchgit, fetchhg, fetchbzr, fetchsvn, ... } @ args:

with goPackages;

buildGoPackage (rec {
  name = "go2nix-${version}";
  # ...
} // (lib.filterAttrs (n: v: n != "lib" && n != "stdenv" && n != "goPackages" && n != "fetchbzr" && n != "fetchgit" && n != "fetchsvn" && n != "fetchhg") args))
```